src/services/identity/CollaboratorsService.ts

```diff
@@ -225,30 +231,51 @@ class CollaboratorsService {
     siteName: string,
     userId: string
   ): Promise<CollaboratorRoles | null> => {
-    const site = await this.siteRepository.findOne({
-      include: [
-        {
-          model: User,
-          as: "site_members",
-          where: {
-            id: userId,
-          },
-        },
-        {
-          model: Repo,
-          where: {
-            name: siteName,
-          },
-        },
-      ],
-    })
+    // Note: We could make the query even cheaper if we assume the site and user cannot be deleted when this verification is done
+    // Basically with just:
+    // SELECT sm.role
+    // FROM
+    //     site_members sm,
+    //     repos r,
+    // WHERE
+    //     r.site_id = sm.site_id
+    //     AND r.name = :siteName
+    //     AND sm.user_id = :userId
+    //     AND r.deleted_at IS NULL
+
+    const records = (await this.sequelize.query(
+      `
+        SELECT sm.role
+        FROM
+            site_members sm,
+            sites s,
+            repos r,
+            users u
+        WHERE
+            r.site_id = s.id
+            AND sm.site_id = s.id
+            AND r.name = :siteName
+            AND sm.user_id = u.id
+            AND sm.user_id = :userId
+            AND r.deleted_at IS NULL
+            AND s.deleted_at IS NULL
+            AND u.deleted_at IS NULL
+      `,
+      {
+        replacements: { siteName, userId },
+        type: QueryTypes.SELECT,
+      }
+    )) as { role: CollaboratorRoles }[]
+
+    if (records.length > 0) {
+      return records[0].role
```

Comment:
for clarification: when does this occur ah? by right this should never be the case right?

Reply:
Hmm? No, it's possible there's no record for a particular user in a particular site, right? 

Exactly in the same sense the previous code was doing this:

```typescript
    return site?.site_members?.[0]?.SiteMember?.role ?? isomerAdminRole
```

II suppose specifically: if an isomer admin visits a site they are not an official collaborator of, there'd be no record in the DB, but `getRole()` will still return `CollaboratorRoles.IsomerAdmin`.

Also if the repo/site/user are soft-deleted, the query will also return an empty set (just like in the previous code)

Reply:
Basically, this is to block access for folks trying to crawl resource they are not supposed to have access to. I mean, it's what `verifySiteMember()` is there to do, right? 😅

Reply:
hmm for clarity records.length should by right = 1 or 0 right? we are just making a statement > 0 defensively correct?

Reply:
Yes, correct!

Reply:
Ah, so you were commenting on the `> 0` ? I thought you were asking about the conditional check itself 😅 

I can make this more explicit with a `LIMIT 1` and doing an explicit check for length 1